app/store.js

```diff
@@ -0,0 +1,20 @@
+const mongoose = require('mongoose');
+const User = require('./model');
+
+module.exports = ({
+    url
+}) => {
+    return mongoose.connect(url, {
+        useNewUrlParser: true
+    }).then((data) => {
+        return {
+            saveUser: (user) => {
+                console.log("A new user has been created.")
```

Comment:
this is a lie - you are logging that you saved the user before you actually did. Better:

```return User.create(user)
.then((response) => {
   console.log("A new user has been created."); // now this is true since we are in the "then" part
   return response; // always do that in case the client calling this function is interested in the result from .create
})```

Reply:
Have done this change.  Never knew you could have .then method inside of a .then - I originally thought they could only come after one another.  Makes a lot of sense however now thinking about it and seeing it.

Reply:
As I explained, this 'then' belongs to the context of the 'saveUser' function. So it will only work when you use that function. In that regard, creating the store has nothing to do with the 'saveUser' function (in terms of execution)